tests/cxx/badinc.cc

```diff
@@ -307,25 +307,20 @@ template<> struct Cc_TemplateStruct<I1_Class, I12> { I1_Class* i; };
 // IWYU: I1_Enum is...*badinc-i1.h
 template<class T = I1_Class, I1_Enum E = I11> class Cc_DeclareOnlyTemplateClass;
 
-// I2_Class has a non-explicit constructor (actually, two), so we need
-// the full type here even though it's a const reference.  I1_Class
-// has no implicit, one-argument constructor, so fwd declaring is ok.
+// I2_Class has non-explicit constructors, so would normally trigger special
```

Comment:
Do we even need to keep this construct in `badinc.cc` now that autocast is covered elsewhere?

Reply:
Not sure if the return-type is only covered here, but it seems unlikely that it wouldn't be tested elsewhere. Returning function static by reference might be worth keeping.

Reply:
I've decided also to keep one passed-by-value parameter in order to test requirement of its type info in function definition. It's been made in #1108.